### PR TITLE
docs: Write a guide for creating a new test

### PR DESCRIPTION
**Pull request by lukehesluke**
And, a guide for Opportunity Criteria

Closes https://github.com/openactive/openactive-test-suite/issues/561 and https://github.com/openactive/openactive-test-suite/issues/564

### CODE REVIEW

File: packages/openactive-integration-tests/test/templates/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Request Templates
+
+TODO
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/templates/README.md:3]: Issue to create this: https://github.com/openactive/openactive-test-suite/issues/566

File: packages/openactive-integration-tests/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+### 1. New feature?
+
+If creating a new feature, create a new folder within `test/features/` with the name of the feature. Within that folder, create a `feature.json` file as described in [feature.json](#featurejson).
+
+If this new test is being added to an existing feature, then skip this step.
+
+### 2. New Test Interface Criteria?
+
+Each test has a set of Criteria configured (see [Opportunity Criteria](#opportunity-criteria)). Check whether or not this test can use a Criteria that already exists or if a new one is needed. If a new Criteria is needed, be sure to create it in the [Test Interface](https://openactive.io/test-interface/) spec, as well as in the [`test-interface-criteria`](../test-interface-criteria/) package.
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/CONTRIBUTING.md:144]: I haven't gone into more detail here about how to create a new one. For the Test Suite part, there is an issue for that here: https://github.com/openactive/openactive-test-suite/issues/554. Not sure if we need documentation for the Test Interface side

File: packages/openactive-integration-tests/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+    If a control isn't appropriate for a particular test, this can be set to the same value as `testOpportunityCriteria`.
+
+This is not an exhaustive list of properties which `FeatureHelper.describeFeature(..)` accepts. See the code documentation for the full picture.
+
+For more information on `FeatureHelper`, see [the ARCHITECTURE doc](./ARCHITECTURE.md#feature-helper).
+
+### Opportunity Criteria
+
+What a Criteria is is documented in [the test-interface-criteria docs](../test-interface-criteria/README.md#criteria).
```
Comment (civsiv) [inline packages/openactive-integration-tests/CONTRIBUTING.md:117]: This reads slightly funny to me, maybe something like:
```suggestion
Each Test Interface Criteria is defined and documented in [the test-interface-criteria docs](../test-interface-criteria/README.md#criteria).
```

File: packages/openactive-integration-tests/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+function (configuration, orderItemCriteriaList, featureIsImplemented, logger) {
+```
+
+An explanation of the main properties:
+
+* `testCategory`, `testFeature`, `testIdentifier`: Identifiers for the **Category**, **Feature** and **Test** respectively. For info on what these are, see [Structure](#structure).
+* `testFeatureImplemented`: Whether or not this test is **implemented** or **non-implemented**. For info on what this means, see [Structure](#structure).
+* `testName`, `testDescription`: Short- and long-form human-readable descriptions of the text. These are used for automatic Documentation Generation and for the test report.
+* `testOpportunityCriteria`: The **Opportunity Criteria** to use for the primary OrderItem under test. For info on criteria, see [Opportunity Criteria](#opportunity-criteria).
+* `controlOpportunityCriteria`: The **Opportunity Criteria** to use for any "control" OrderItems.
```
Comment (civsiv) [inline packages/openactive-integration-tests/CONTRIBUTING.md:103]:
```suggestion
* `controlOpportunityCriteria`: The **Opportunity Criteria** to use for any "control" OrderItems. For info on criteria, see [Opportunity Criteria](#opportunity-criteria).
```

Comment (civsiv) [reply packages/openactive-integration-tests/CONTRIBUTING.md:103]: A dev might not necessarily read the line about `testOpportunityCriteria`

File: packages/openactive-integration-tests/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+2. **C1-and-Assert-Capacity**: Calls C1 for the selected Opportunity-Offer combos. Additionaly, checks, via the Broker Microservice, that the capacity of the Opportunity is as expected after calling C1 (it should be unchanged unless anonymous leasing is supported).
+3. **C2-and-Assert-Capacity**: Similar to C1 but for C2.
+4. **Book-and-Assert-Capacity**: How this behaves depends on whether Simple Booking Flow or Approval Booking Flow is under test (See: [Test Matrix](./ARCHITECTURE.md#test-matrix)):
+    * **Simple Booking Flow**: Similarly to C1 and C2, calls B and checks that the capacity is as expected.
+    * **Approval Booking Flow**: Calls: P -> Order Proposal Feed Update -> B, and checks that the capacity is as expected after each of P and B.
+
+If you need to test a different flow from this, there are many other Flow Stage Recipes available, and it's always possible to create custom flows from scratch.
+
+Note that the Flow Stages are just initiated. None of the API calls are made yet.
```
Comment (civsiv) [inline packages/openactive-integration-tests/CONTRIBUTING.md:189]:
```suggestion
Note that the Flow Stages are just initiated during set up. None of the API calls are made yet.
```

File: packages/test-interface-criteria/README.md [hunk 6]
```diff
@@ -21,6 +21,44 @@ const TestOpportunityBookableCriteria = TestInterfaceCriteria.criteriaMap.get('T
 const { matchesCriteria, unmetCriteriaDetails } = TestInterfaceCriteria.testMatch(TestOpportunityBookableCriteria, opportunity);
 ```
 
+## Concepts
```
Comment (civsiv) [inline packages/test-interface-criteria/README.md:24]: Thoughts on this section being above Usage?

**Review by civsiv — APPROVED** [review]

File: packages/openactive-integration-tests/CONTRIBUTING.md (hunk 4 above)
Comment (lukehesluke) [reply packages/openactive-integration-tests/CONTRIBUTING.md]: Great point. Done

File: packages/openactive-integration-tests/CONTRIBUTING.md (hunk 5 above)
Comment (lukehesluke) [reply packages/openactive-integration-tests/CONTRIBUTING.md:189]: Yes! Done

File: packages/test-interface-criteria/README.md (hunk 6 above)
Comment (lukehesluke) [reply packages/test-interface-criteria/README.md:24]: My rationale was that it seems to me to be a common pattern for usage/examples to come first in project READMEs. Perhaps geared towards people who are already somewhat knowledgeable and want to get stuck in

**Pull request merged**